test/e2e/cluster/add_remove_cluster.go

```diff
@@ -0,0 +1,41 @@
+package cluster
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/submariner-io/submariner/test/e2e/dataplane"
+	"github.com/submariner-io/submariner/test/e2e/framework"
+)
+
+var _ = Describe("Test to check adding/removing a new cluster to existing setup", func() {
+	f := framework.NewDefaultFramework("add-remove-cluster")
+
+	It("Should be able to add and remove 3rd cluster", func() {
+		By(fmt.Sprintf("Verifying that initially pods from the other clusters can not communicate to service/pod in cluster 1"))
+
+		dataplane.RunNoConnectivityTest(f, false, framework.NonGatewayNode, framework.NonGatewayNode, framework.ClusterA, framework.ClusterC)
+		dataplane.RunNoConnectivityTest(f, true, framework.GatewayNode, framework.NonGatewayNode, framework.ClusterB, framework.ClusterC)
+
+		By(fmt.Sprintf("Adding a new cluster to the setup by setting gateway label to one of the non gateway nodes on cluster 1"))
+		nonGatewayNode := f.FindNodesByGatewayLabel(framework.ClusterC, false)[0].Name
+		f.SetGatewayLabelOnNode(framework.ClusterC, nonGatewayNode, true)
+		gatewayNodes := f.FindNodesByGatewayLabel(framework.ClusterC, true)
+		framework.Logf("gateway node is %q", gatewayNodes[0].Name)
+		Expect(gatewayNodes).To(HaveLen(1), fmt.Sprintf("Expected only one gateway node on %q", framework.ClusterC))
```

Comment:
I don't think lines 24-26 are necessary.

Reply:
Its a sanity check that GW node exists.

Reply:
But it doesn't really add any value since we just successfully set the label on `nonGatewayNode` other than to verify correctness of the `SetGatewayLabelOnNode` function but I don't think that's necessary here.

Reply:
Done